packages/@aws-cdk/aws-ecs/lib/base/base-service.ts

```diff
@@ -68,6 +68,39 @@ export interface DeploymentCircuitBreaker {
   readonly rollback?: boolean;
 }
 
+/**
+ * Alarm behavior of an ECS service
+ */
+export enum AlarmBehavior {
+  /**
+   * ROLLBACK_ON_ALARM causes the service to roll back to the previous deployment
+   * when any configured alarm is in the 'Alarm' state. The Cloudformation stack
+   * will be rolled back and enter state "UPDATE_ROLLBACK_COMPLETE".
+   */
+  ROLLBACK_ON_ALARM = 0,
+  /**
+   * FAIL_ON_ALARM causes the deployment to fail immediately. In order to restore
+   * functionality, you must roll the stack forward by pushing a new version of the
+   * ECS service.
+   */
+  FAIL_ON_ALARM = 1,
+}
+
+/**
+ * The deployment alarms to use for the service
+ */
+export interface DeploymentAlarmConfig {
+  /**
+   * Default rollback on alarm
+   * @default AlarmBehavior.ROLLBACK_ON_ALARM
+   */
+  readonly behavior?: AlarmBehavior;
+  /**
+   * List of alarms
+   */
+  readonly alarms: cloudwatch.IAlarm[];
```

Comment:
I'm curious if this can be an empty array - what happens if we set alarm config with `alarms: []` ?

Reply:
This validation will be added in subsequent PR , so alarms should be minimum one.